recipes/default.rb

```diff
@@ -34,26 +34,91 @@
   not_if { node["kagent"]["enabled"] == "false" }
 end
 
-# Generate an API key 
-api_key = ""
+# Generate an API key
+ruby_block 'fetch-api-key' do
+  block do
+    require 'net/https'
+    require 'http-cookie'
+    require 'json'
 
-# Template the configuration file 
+    hopsworks_fqdn = consul_helper.get_service_fqdn("hopsworks.glassfish")
+    hopsworks_endpoint = "https://#{hopsworks_fqdn}:#{node['hopsworks']['internal']['port']}"
+    url = URI.parse("#{hopsworks_endpoint}/hopsworks-api/api/auth/service")
+    api_key_url = URI.parse("#{hopsworks_endpoint}/hopsworks-api/api/users/apiKey")
+
+    params =  {
+      :email => node['onlinefs']['hopsworks']['email'],
+      :password => node['onlinefs']['hopsworks']["password"]
+    }
+
+    api_key_params = {
+      :name => "onlinefs",
+      # check with fabio which scopes are needed
+      :scope => "FEATURESTORE,KAFKA,PROJECT"
+    }
+
+    http = Net::HTTP.new(url.host, url.port)
+    http.read_timeout = 120
+    http.use_ssl = true
+    http.verify_mode = OpenSSL::SSL::VERIFY_NONE
+
+    jar = ::HTTP::CookieJar.new
+
+    http.start do |connection|
+
+      request = Net::HTTP::Post.new(url)
+      request.set_form_data(params, '&')
+      response = connection.request(request)
+
+      if( response.is_a?( Net::HTTPSuccess ) )
+          # your request was successful
+          puts "The Response -> #{response.body}"
+
+          response.get_fields('Set-Cookie').each do |value|
+            jar.parse(value, url)
+          end
+
+          request = Net::HTTP::Post.new(api_key_url)
+          request.set_form_data(api_key_params, '&')
+          request['Content-Type'] = "application/json"
+          request['Cookie'] = ::HTTP::Cookie.cookie_value(jar.cookies(api_key_url))
+          request['Authorization'] = response['Authorization']
+          response = connection.request(request)
+
+          if ( response.is_a? (Net::HTTPSuccess))
+            json_response = ::JSON.parse(response.body)
+            api_key = json_response['signedCert']
```

Comment:
I think you still need to define this outside the `ruby_block` otherwise chef won't find it when compiling.

Reply:
fair enough